crates/task-impls/src/transactions.rs

```diff
@@ -92,28 +93,42 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
     ) -> Option<HotShotTaskCompleted> {
         match event {
             HotShotEvent::TransactionsRecv(transactions) => {
-                let consensus = self.consensus.read().await;
-                self.transactions
-                    .modify(|txns| {
-                        for transaction in transactions {
-                            let size = bincode_opts().serialized_size(&transaction).unwrap_or(0);
-
-                            // If we didn't already know about this transaction, update our mempool metrics.
-                            if !self.seen_transactions.remove(&transaction.commit())
-                                && txns.insert(transaction.commit(), transaction).is_none()
-                            {
-                                consensus.metrics.outstanding_transactions.update(1);
-                                consensus
-                                    .metrics
-                                    .outstanding_transactions_memory_size
-                                    .update(i64::try_from(size).unwrap_or_else(|e| {
-                                        warn!("Conversion failed: {e}. Using the max value.");
-                                        i64::MAX
-                                    }));
-                            }
-                        }
-                    })
-                    .await;
+                futures::join! {
+                    self.api
+                        .send_event(Event {
+                            view_number: self.cur_view,
+                            event: EventType::Transactions {
+                                transactions: transactions.clone(),
+                            },
+                        }),
```

Comment:
And here

Reply:
Okay, I think I remembered why I did this duplication. For transactions I feel it still makes sense, because if the sender isn't in DA committee, it won't receive the transaction back. That was the first case I implemented and just did the same for other events by inertia. I'll fix the other ones obviously, but for transactions it's justified I think.

Reply:
okay, this makes sense.  We should really fix the networking so that transactions go to everyone and not the DA committee, but we can do that after the new CDN is finished.